SPARQLWrapper/Wrapper.py

```diff
@@ -557,70 +566,63 @@ def setHTTPAuth(self, auth):
-            pass
+        if not isinstance(query, (str, bytes)):
+            raise TypeError(type(query))
         elif isinstance(query, bytes):
             query = query.decode("utf-8")
-        else:
-            raise TypeError(
-                "setQuery takes either unicode-strings or utf-8 encoded byte-strings"
-            )
 
         self.queryString = query
         self.queryType = self._parseQueryType(query)
 
-    def _parseQueryType(self, query):
+    def _parseQueryType(self, query: Union[str, bytes]) -> Optional[str]:
         """
-            Internal method for parsing the SPARQL query and return its type (ie, :data:`SELECT`, :data:`ASK`, etc).
+        Internal method for parsing the SPARQL query and return its type (ie, :data:`SELECT`, :data:`ASK`, etc).
 
-            .. note::
-              The method returns :data:`SELECT` if nothing is specified. This is just to get all other
-              methods running; in fact, this means that the query is erroneous, because the query must be,
-              according to the SPARQL specification. The
-              SPARQL endpoint should raise an exception (via :mod:`urllib`) for such syntax error.
+        .. note::
+          The method returns :data:`SELECT` if nothing is specified. This is just to get all other
+          methods running; in fact, this means that the query is erroneous, because the query must be,
+          according to the SPARQL specification. The
+          SPARQL endpoint should raise an exception (via :mod:`urllib`) for such syntax error.
 
-            :param query: query text.
-            :type query: string
-            :return: the type of SPARQL query (aka SPARQL query form).
-            :rtype: string
+        :param query: query text.
+        :type query: string
+        :return: the type of SPARQL query (aka SPARQL query form).
+        :rtype: string
         """
-        try:
-            query = (
-                query if (isinstance(query, str)) else query.encode("ascii", "ignore")
-            )
-            query = self._cleanComments(query)
-            query_for_queryType = re.sub(self.prefix_pattern, "", query.strip())
-            r_queryType = (
-                self.pattern.search(query_for_queryType).group("queryType").upper()
-            )
-        except AttributeError:
+        query = query if isinstance(query, str) else query.decode("utf-8")
+        query = self._cleanComments(query)
+        query_for_queryType = re.sub(self.prefix_pattern, "", query.strip())
+        m = self.pattern.search(query_for_queryType)
+        if m is None:
             warnings.warn(
                 "not detected query type for query '%s'" % query.replace("\n", " "),
                 RuntimeWarning,
             )
             r_queryType = None
+        else:
+            r_queryType = m.group("queryType").upper()
 
-        if r_queryType in _allowedQueryTypes:
+        if isinstance(r_queryType, str) and r_queryType in _allowedQueryTypes:
```

Comment:
This looks like it can go into a separate PR.

Reply:
934e334